release-26.2: ccl/changefeedccl: TestChangefeedOrderingWithErrors failed

ccl/changefeedccl.TestChangefeedOrderingWithErrors [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21290425?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21290425?buildTab=artifacts#/) on release-26.2 @ [9469efe4ec4e643dc72754e93aa3642f0510d95d](https://github.com/cockroachdb/cockroach/commits/9469efe4ec4e643dc72754e93aa3642f0510d95d):


```
=== RUN   TestChangefeedOrderingWithErrors
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/caa72099da386999b346f93d4299a93a/logTestChangefeedOrderingWithErrors3131451559
    test_log_scope.go:82: use -show-logs to present logs inline
    changefeed_test.go:9643: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/caa72099da386999b346f93d4299a93a/logTestChangefeedOrderingWithErrors3131451559
--- FAIL: TestChangefeedOrderingWithErrors (9.99s)
=== RUN   TestChangefeedOrderingWithErrors/webhook
    helpers_test.go:1576: making server as secondary tenant
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    helpers_test.go:1655: making webhook feed factory
    changefeed_test.go:9628: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:9628
        	            				pkg/testutils/soon.go:76
        	            				pkg/util/retry/retry.go:362
        	            				pkg/testutils/soon.go:82
        	            				pkg/testutils/soon.go:59
        	            				pkg/testutils/soon.go:38
        	            				pkg/ccl/changefeedccl/changefeed_test.go:9623
        	            				pkg/ccl/changefeedccl/helpers_test.go:1841
        	            				pkg/ccl/changefeedccl/helpers_test.go:1877
        	Error:      	Expect "starting streams (1 out of 1)" to match "500 Internal Server Error"
        	Test:       	TestChangefeedOrderingWithErrors/webhook
--- FAIL: TestChangefeedOrderingWithErrors/webhook (9.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOrderingWithErrors.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62236

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -9649,14 +9649,17 @@ func TestChangefeedOrderingWithErrors(t *testing.T) {
 		sqlDB.Exec(t, `UPSERT INTO foo VALUES (1, 'd')`)
 		feedJob := foo.(cdctest.EnterpriseTestFeed)
 
-		// check that running status correctly updates with retryable error
+		// Check that running status correctly updates with retryable error.
 		testutils.SucceedsSoon(t, func() error {
-			status, err := feedJob.FetchStatusMessage()
-			if err != nil {
-				return err
-			}
-			require.Regexp(t, "500 Internal Server Error", status)
-			return nil
+			var unused int
+			// If no rows are found by QueryRow, Scan will return an error (ErrNoRows)
+			return s.DB.QueryRow(`
+				SELECT 1 FROM system.job_message
+				WHERE job_id = $1
+				  AND kind = 'status'
+				  AND message LIKE '%500 Internal Server Error%'
+				LIMIT 1
+			`, feedJob.JobID()).Scan(&unused)
 		})
 
 		webhookFoo.mockSink.SetStatusCodes([]int{http.StatusOK})

